parser/src/main/scala/org/globalnames/parser/Parser.scala

```diff
@@ -8,11 +8,51 @@ import scalaz.Scalaz._
 
 object Parser extends org.parboiled2.Parser {
 
-  class Context(val parserWarnings: ParserWarnings)
+  class Context(val parserWarnings: ParserWarnings,
+                val preprocessChanges: Boolean)
+
+  val sciCharsExtendedSet = "æœſàâåãäáçčéèíìïňññóòôøõöúùüŕřŗššşž'".toSet
+  val sciUpperCharExtendedSet = "ÆŒ".toSet
+  val authCharUpperSet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".toSet |
+    "ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝĆČĎİĶĹĺĽľŁłŅŌŐŒŘŚŜŞŠŸŹŻŽƒǾȘȚ".toSet
+  val authCharMiscodedSet = Set('�')
+  val apostrSet = Set('\'')
+  val spaceMiscodedSet = "　 \t\r\n\f_".toSet
+  val doubleSpacePattern = """[\s]{2,}""".r
 
   val sciName: Rule1[ScientificName] = rule {
-    softSpace ~ sciName1 ~ anyChars ~ EOI ~> {(ng: NamesGroup, garbage: String) =>
-      ScientificName(namesGroup = ng.some, garbage = garbage)
+    capturePos(softSpace ~ sciName1) ~ anyChars ~ EOI ~> {
+      (ng: NamesGroup, pos: CapturePos, garbage: String) =>
+      val name = state.input.sliceString(pos.start, pos.end)
+      name match {
+        case n if doubleSpacePattern.findFirstIn(n) != None =>
+          ctx.parserWarnings
+            .add(Warning(2, "Some spaces between words are not single", ng))
+        case _ =>
+      }
+      name.toSet match {
```

Comment:
@dimus  `name.toSet` could have non-empty intersection with `spaceMiscodedSet` and `authCharMiscodedSet` simultaneosly, right?